application.cpp

```diff
@@ -121,6 +123,22 @@ void application::startup() {
    }
 }
 
+void application::start_sighup_handler(std::shared_ptr<boost::asio::io_service> sig_io_serv) {
+   cout << "Starting sighup handler" << std::endl;
+   std::shared_ptr<boost::asio::signal_set> sighup_set(new boost::asio::signal_set(*sig_io_serv, SIGHUP));
+   sighup_set->async_wait([sig_io_serv, sighup_set, this](const boost::system::error_code& err, int /*num*/) {
+      cout << "Invoking sighup handler" << std::endl;
+      if(!err) {
+         sighup_callback();
+         for( auto plugin : initialized_plugins ) {
+            if( is_quiting() ) return;
+            plugin->initialize_logging();
```

Comment:
I think this should be `plugin->handle_sighup()` so a plugin can respond to sighup anyway it wants which would include initializing logging. doxygen will also need to be updated.